packages/payer-logo/PayerLogo.js

```diff
@@ -27,18 +28,66 @@ query($payerIDs: [String!], $types: [String!]){
 }
 `;
 
-export default class PayerLogo extends Component {
-  state = {};
+const getLogoBySpaceId = async spaceId =>
+  new Promise(async (resolve, reject) => {
+    try {
+      const {
+        data: {
+          data: { space },
+        },
+      } = await avSlotMachineApi.create({
+        query: spaceIDQuery,
+        variables: { id: spaceId },
+      });
+
+      const images = get(space, 'images', []).reduce(
+        (accum, { name, value }) => {
+          accum[name] = value;
+          return accum;
+        },
+        {}
+      );
+
+      return resolve(images.logo);
+    } catch (error) {
+      return reject(error);
+    }
+  });
+
+const getLogoByPayerId = async payerId =>
+  new Promise(async (resolve, reject) => {
+    try {
+      const {
+        data: {
+          data: { spaces },
+        },
+      } = await avSlotMachineApi.create({
+        query: payerIDQuery,
+        variables: { payerIDs: [payerId], types: ['space'] },
+      });
+
+      const images = get(spaces, '[0].images', []).reduce(
+        (accum, { name, value }) => {
+          accum[name] = value;
+          return accum;
+        },
+        {}
+      );
 
-  async componentDidMount() {
-    const { spaceId, payerId } = this.props;
+      return resolve(images.logo);
+    } catch (error) {
+      return reject(error);
+    }
+  });
 
+const getLogo = (spaceId, payerId) =>
+  new Promise(async (resolve, reject) => {
     try {
       let url;
       if (spaceId) {
-        url = await this.getLogoBySpaceId();
+        url = await getLogoBySpaceId(spaceId);
```

Comment:
There should be a way you can just create 1 function out of this and pass in a simple object with params so we don't have a lot of duplicate logic in both `GET` methods.

Reply:
please see 337253dfc06c9b6fbdedcdbd31b782e696f64f55